lib/ansible.js

```diff
@@ -1,106 +1,131 @@
-var shelljs = require('shelljs');
+var exec = require('child_process');
 var _ = require('underscore');
-var format = require('util').format;
 var inherits = require('util').inherits;
 var utils = require('./utils');
 var Q = require('q');
+var EventEmitter = require('events').EventEmitter;
+
+var AbstractAnsibleCommand = function() {
+  EventEmitter.call(this);
+};
+
+inherits(AbstractAnsibleCommand, EventEmitter);
+
+AbstractAnsibleCommand.prototype.exec = function(options) {
+  // Validate execution configuration
+  var errors = this.validate();
+  var self = this;
+  if (errors.length > 0) {
+    var error = new Error('Ansible execution was mis-configured');
+    error.reason = errors;
+    return Q.reject(error)
+  }
 
-var AbstractAnsibleCommand = function() {}
+  var deferred = Q.defer();
 
-AbstractAnsibleCommand.prototype = {
+  var processOptions = {};
+  if (options && options.cwd) {
+    processOptions.cwd = options.cwd;
+  }
 
-  exec: function(options) {
-    // Validate execution configuration
-    var errors = this.validate();
-    if (errors.length > 0) {
-      var error = new Error('Ansible execution was mis-configured');
-      error.reason = errors;
-      return Q.reject(error)
-    }
+  var output = '';
+  var child = exec.spawn(this.commandName(), this.compileParams(), processOptions);
 
-    var deferred = Q.defer();
+  child.stdout.on('data', function(data) {
+    var dataString = data.toString();
```

Comment:
It does make sense. I had it toString because of my current use case, but keeping raw is better. Fixed here and in the documentation.